server/src/test/java/org/elasticsearch/index/IndexSettingProviderTests.java

```diff
@@ -23,17 +23,25 @@
 public class IndexSettingProviderTests extends ESSingleNodeTestCase {
 
     public void testIndexCreation() throws Exception {
-        var indexService = createIndex("my-index1");
+        Settings settings = Settings.builder().put("index.mapping.depth.limit", 10).build();
+        var indexService = createIndex("my-index1", settings);
         assertFalse(indexService.getIndexSettings().getSettings().hasValue("index.refresh_interval"));
+        assertEquals("10", indexService.getIndexSettings().getSettings().get("index.mapping.depth.limit"));
 
         INDEX_SETTING_PROVIDER1_ENABLED.set(true);
-        indexService = createIndex("my-index2");
+        indexService = createIndex("my-index2", settings);
         assertTrue(indexService.getIndexSettings().getSettings().hasValue("index.refresh_interval"));
+        assertEquals("10", indexService.getIndexSettings().getSettings().get("index.mapping.depth.limit"));
+
+        INDEX_SETTING_OVERRULING.set(true);
+        indexService = createIndex("my-index3", settings);
+        assertTrue(indexService.getIndexSettings().getSettings().hasValue("index.refresh_interval"));
+        assertEquals("100", indexService.getIndexSettings().getSettings().get("index.mapping.depth.limit"));
 
         INDEX_SETTING_PROVIDER2_ENABLED.set(true);
-        var e = expectThrows(IllegalArgumentException.class, () -> createIndex("my-index3"));
+        var e = expectThrows(IllegalArgumentException.class, () -> createIndex("my-index4", settings));
         assertEquals(
-            "additional index setting [index.refresh_interval] added by [TestIndexSettingsProvider] is already present",
```

Comment:
I think this needs to be updated with the new setting in this test. However the order in which the settings keys are returned is non deterministic (keys in `Settings` is backed by a hash set). Maybe just adjust the assert to either expect to error messages or just check the the error message ends with ` added by [TestIndexSettingsProvider] is already present`?

Reply:
Yeah noticed it's flaky. Should be fixed now.